### PR TITLE
Style engine: update type for getCSSValueFromRawStyle

### PR DESCRIPTION
**Pull request by ramonjd**
## What?

Follow up to https://github.com/WordPress/gutenberg/pull/64490

Use a generic type parameter `StyleValue` to describe argument and return value of `getCSSValueFromRawStyle()`.

Props to @kevin940726 

## Why?

Avoid the use of `any` and describe better what the function expects.

## How?

By default, `StyleValue` is set to string. This allows the function to be flexible with the type of styleValue, but it defaults to handling strings.

`getCSSValueFromRawStyle()` takes one parameter, `styleValue`, which is of type `StyleValue = string`. This means it can be any type, but the function is primarily designed to work with strings.

`getCSSValueFromRawStyle()` returns a value of the same type `StyleValue`. This ensures that its output type matches its input type.

## Testing Instructions

The CI should build, all tests should pass.

### CODE REVIEW

File: packages/style-engine/src/styles/utils.ts [hunk 1]
```diff
@@ -139,9 +141,9 @@ export function generateBoxRules(
  * @return A CSS var value.
  */
 
-export function getCSSValueFromRawStyle(
-	styleValue: string | any
-): string | unknown {
+export function getCSSValueFromRawStyle< CSSStyleValue = string >(
```
Comment (kevin940726) [inline packages/style-engine/src/styles/utils.ts]: `CSSStyleValue` is a [defined type](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleValue) which could be a little bit confusing for others 😅.

Comment (ramonjd) [reply packages/style-engine/src/styles/utils.ts]: 🤦🏻 **TIL**

I need to call it something. `T` in the docs is a bit confusing. 

`RawStyleValue` or something

Comment (kevin940726) [reply packages/style-engine/src/styles/utils.ts]: Or Just `StyleValue` or even `Type`?

Comment (ramonjd) [reply packages/style-engine/src/styles/utils.ts]: Changed to `StyleValue` so it looks good in the generated docs.

File: packages/style-engine/README.md [hunk 2]
```diff
@@ -260,19 +260,21 @@ _Changelog_
...
+The preset value is a string and follows the pattern `var:description|context|slug`.
 
 Example:
 
 `getCSSValueFromRawStyle( 'var:preset|color|heavenlyBlue' )` // returns 'var(--wp--preset--color--heavenly-blue)'
 
 _Parameters_
 
--   _styleValue_ `string | any`: A raw style value.
+-   _styleValue_ `StyleValue`: A raw style value.
```
Comment (kevin940726) [inline packages/style-engine/README.md]: Nit: We could add a simple wording here explaining this is mostly going to be a string.

**Review by kevin940726 — APPROVED** [review]
LGTM 👍

File: packages/style-engine/README.md (hunk 2 above)
Comment (ramonjd) [reply packages/style-engine/README.md]: Will add, thanks again!

**Pull request merged**